validator/validate.go

```diff
@@ -0,0 +1,106 @@
+package validator
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gorilla/schema"
+)
+
+// ErrFormValidationFailed if form validation has failed, if this is ever returned,
+// corresponding call to GetFieldErrors should be made to check for individual field errors
+var ErrFormValidationFailed = errors.New("form validation failed, check field errors")
+
+// FormValidator represents a form validator with appropriate validator method
+type FormValidator struct {
+	rulesJSON []byte
```

Comment:
we should consider how we could make this goroutine-safe? i.e. so `FormValidator` can be created once with the rules JSON and reused across multiple forms/requests?

currently it'd require parsing the rules JSON on every request